python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,266 @@
+    return model
+
+
+def generate(args):
+    """
+    create an ml4ir consumable dataframe after generating dataset.
+    """
+    X, Y = generate_data(args)
+    df = create_ml4ir_dataframe(X, Y)
+    return df
+
+
+def wrap_linear_model_in_sklearn_model(df, W):
+    """
+    Wrap plain weights and bias in a dummy sklearn model
+    """
+    wrapper_model = Perceptron(fit_intercept=True, max_iter=1, verbose=0, validation_fraction=0.0001)
+    X = [df['f' + str(i)] for i in range(len(W)-1)]
+    Y = df['clicked']
+    wrapper_model.fit(np.array(X).T, Y)
+    for i in range(len(W)-1):
+        wrapper_model.coef_[0][i] = W[i]
+    wrapper_model.intercept_ = W[-1]
+    return wrapper_model
+
+
+def test_ml4ir_sanity_pipeline(args, p, log_dir):
+        df = generate(args)
+        df.to_csv(p / 'train' / 'data.csv')
+        df.to_csv(p / 'validation' / 'data.csv')
+        df.to_csv(p / 'test' / 'data.csv')
+
+        fconfig_name = "feature_config_sanity_tests_" + str(args.n_features) + "_features.yaml"
+        feature_config_file = pathlib.Path(__file__).parent / "data" / "configs" / fconfig_name
+        model_config_file = pathlib.Path(__file__).parent / "data" / "configs" / "model_config_sanity_tests.yaml"
+        train_ml4ir(p.as_posix(), feature_config_file.as_posix(), model_config_file.as_posix(), log_dir.as_posix())
+        logisticReg = LogisticRegression(fit_intercept=False, random_state=args.seed, max_iter=100,
+                                warm_start=True)
+        perceptron = Perceptron(fit_intercept=False, max_iter=100, random_state=args.seed, validation_fraction=0.0001,
+                       warm_start=True)
+        ml4ir_weights = pd.read_csv("models/test_command_line/coefficients.csv")["weight"].tolist()
+        ml4ir_weights.append(0.0)
+        ml4ir = wrap_linear_model_in_sklearn_model(df, ml4ir_weights)
+        train_sklearn_model(df, logisticReg, featureCount=args.n_features)
+        train_sklearn_model(df, perceptron, featureCount=args.n_features)
+
+        ml4ir_mrr = check_MRR(df, ml4ir, args.n_features)
+        perceptron_mrr = check_MRR(df, perceptron, args.n_features)
+        log_regression_mrr = check_MRR(df, logisticReg, args.n_features)
+        return ml4ir_mrr, perceptron_mrr, log_regression_mrr
+
+
+def train_ml4ir(data_dir, feature_config, model_config, logs_dir):
+
+    argv = ['--data_dir', data_dir, '--feature_config', feature_config, '--loss_type', "pointwise", '--scoring_type', "pointwise",
+            '--run_id', 'test_command_line', '--data_format', 'csv', '--execution_mode', 'train_inference_evaluate', '--loss_key', 'sigmoid_cross_entropy',
+            '--num_epochs', '150', '--model_config', model_config, '--batch_size', '16', '--logs_dir', logs_dir,
+            '--max_sequence_size', '25', '--train_pcent_split', '1.0', '--val_pcent_split', '-1', '--test_pcent_split', '-1', '--early_stopping_patience', '1000']
+    args = get_args(argv)
+    rp = RankingPipeline(args=args)
```

Comment:
@mohazahran Conversely, if you do not want to do this string list thing, you could directly call `RankingModel` as well, but you'd probably need some additional code. I don't feel too strongly about this as this is also one of the intended use cases of ml4ir.